### ISSUE TITLE
Something went wrong with the screensaver

### ISSUE THREAD

**Issue by kendoori** [linuxmint/cinnamon-screensaver#392]
```
 * Cinnamon-Screensaver version (cinnamon-screensaver --version): cinnamon-screensaver 5.0.6
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) Linux Mint 20.2
 * Graphics hardware *and* driver used: Intel Corporation UHD Graphics 620
 * 32 or 64 bit: 64-bit
 ```

**Issue**
Return to machine after overnight (no powersave suspend) and machine displays the "Something went wrong with the screensaver" message. This started happening after the 20.2 upgrade.

![PXL_20210802_123858264](https://user-images.githubusercontent.com/2447420/127864255-ce7f2d6e-cf09-45f4-a077-bf16a03acfad.jpg)

**Steps to reproduce**
Allow screensaver control overnight. No user input or activity.

**Expected behaviour**
Lock screen should appear

**Other information**
![ksnip_20210802-084925](https://user-images.githubusercontent.com/2447420/127864639-ba2cc454-9ca2-42a7-b396-44a8f53718f8.png)

**Comment by mtwebster** [comment]
Does this happen always or occasionally? Is the monitor 'sleeping' when you come back to unlock it (screen completely blank) or is it already displaying that message?

Please run mintreport ("System Reports" in the menu) and click on the 'Crash reports' tab.
![image](https://user-images.githubusercontent.com/262776/129231251-5216af8b-a56f-4ff4-bc7a-29f2b673b1be.png)
If there are any entries for cinnamon-screensaver please select them (it may take a few moments for the view to update) and then  click the 'Local files' button and attach the 'crash.tar.gz' file to a response here (you can simply drag the file into the typing box on github).

If there is more than one instance of cinnamon-screensaver in the crash list please attach that file from each of them.

Thanks

**Comment by kendoori** [comment]
What happens to me is that I leave the machine idle overnight and this happens during that time. I don't suspend. when I come back, there is the "something went wrong" message active on the screen.

Github is not allowing me to attach the crash report
![Selection_223](https://user-images.githubusercontent.com/2447420/129236182-25e8958a-5879-4871-a28a-ba2bdd16c8f3.png)

Even if I recompress with .zip as suffix it doesn't work.

Here is the pastebin:

https://termbin.com/xi9z

**Comment by mtwebster** [comment]
Hi, can you try uploading it here? https://filebin.net/ and share the download link.

The pastebin you shared was for a 'cinnamon' crash, not 'cinnamon-screensaver', by the way.

Thanks - I know this is tedious. Github does allow .tar.gz files, but it has a 25mb size limit, unfortunately.

**Comment by kendoori** [comment]
Forgive me for being thick. I don't understand how to only show the cinnamon-screensaver crashes. Are you saying that after one of the screen saver-related crashes, I should see a specific crash report for cinnamon-screensaver in the list? if that's the case, I don't show that ever. If that's not the case how would I pinpoint those events?

Here's what I see:

![System Reports_001](https://user-images.githubusercontent.com/2447420/129617126-1ebe66f3-0650-4377-8686-65cd307812f3.png)

**Comment by MarkusAJacobsen** [comment]
I'm also experiencing this issue after upgrading from 20.1 to 20.2. Been using mint for 7 years and this is the first time I'm having this issue.
It usually happens when I boot the pc after having it in hibernate, no matter the timeframe. I've also experienced it on reboot

**Comment by Boored-Dude** [comment]
this error occured on my machine 4 times since Friday (20. August). I have installed all updates that there available (Mint 20.2).
On another machine (all available updates installed) it occured once yesterday.
It never happened on ever machine before.
Steps to reproduce: do nothing till the screensaver turns on.
[crash.tar.gz](https://github.com/linuxmint/cinnamon-screensaver/files/7027020/crash.tar.gz)
[inxi.tar.gz](https://github.com/linuxmint/cinnamon-screensaver/files/7027024/inxi.tar.gz)
[Stack Tracking on Pastebin](https://termbin.com/ndbsi)

**Comment by mtwebster** [comment]
I've come to a bit of a realization - the screensaver here is having issues *because* cinnamon is crashing. I'm going to fix this and get an update out asap, and try to work out the cinnamon issue (which I think we're already tracking).

Thank you for your patience here.

**Comment by dhdurgee** [comment]
I just encountered this for the first time last evening.  I had just enabled the cinnamon memory-limit feature and based on the logs it appears that it attempted a restart overnight that was not completed until after I addressed this issue.  I am attaching what I believe are the relative logs.  If you need more information please let me know.  There was no crash report as cinnamon did not crash, it apparently was improperly restarted.

**Comment by dhdurgee** [comment]
I am now going back to my script. Once again screensaver encountered a problem likely due to their memory-limit restart. Once again had to use Ctrl-Alt-F2 to login and issue "cinnamon-unlock-desktop" so that cinnamon would restart when switched back to GUI.  I have attached this morning's logs, which are similar to yesterday morning.


[restarts.zip](https://github.com/linuxmint/cinnamon-screensaver/files/7072426/restarts.zip)

**Comment by mtwebster** [comment]
Hi - I've got a fix for this committed, I was just trying to get a release out.

If you're willing to test it, you can either:

- download packages.tar.gz from here: (Mint 20) https://github.com/linuxmint/cinnamon-screensaver/releases/tag/master.mint20 or (LMDE 4) https://github.com/linuxmint/cinnamon-screensaver/releases/tag/master.lmde4 and unpack to a folder. Only unpack cinnamon-screensaver and libcscreensaver0 (unless you've previously installed libcscreensaver-dbg for some reason).
- Open a terminal in that folder. run `dpkg -i *.deb`

or

- Download this file: https://raw.githubusercontent.com/linuxmint/cinnamon-screensaver/master/src/stage.py.
- go to your download folder in a terminal
- run `sudo cp stage.py /usr/share/cinnamon-screensaver`

Note: This should fix the screensaver becoming unresponsive/breaking. But, it won't fix cinnamon crashing, if that was the underlying trigger for this.

**Comment by dhdurgee** [comment]
Thank you for your timely response.  I might give this a try later, but at present I have fallen back to an alternative to the memory-limit feature which I believe the cause of this problem that does not appear to result in an improper cinnamon restart.

As I noted in my posts above, I believe the source of the problem is the cinnamon memory-limit causes an improper restart when the screensaver is active.  I believe the logs I have posted here indicate this and there is no crash of cinnamon, simply a failure to restart until the "cinnamon-unlock-desktop" command is issued from a terminal session.

I have also posted about this in issue #10287 for cinnamon.  I include there a copy of the script I am using for restart which does not have this problem.  It uses xdotool to activate the hot-key to request a cinnamon restart and appears to work fine.

**Comment by dhdurgee** [comment]
I have just encountered something that might increase the probability that I will install your screensaver fix.

As your memory-limit function was logging its actions in ~/.cinnamon/restarts.log I decided to edit my alternative script to log its actions as well.  This lead to a surprise, this morning.  I had expected to see a logged restart overnight, instead I found:

Mon 30 Aug 2021 12:48:09 AM EDT restarting cinnamon.
Mon 30 Aug 2021 03:06:17 AM EDT restarting cinnamon.
Mon 30 Aug 2021 03:40:19 AM EDT restarting cinnamon.
Mon 30 Aug 2021 04:03:01 AM EDT restarting cinnamon.
Mon 30 Aug 2021 04:36:02 AM EDT restarting cinnamon.
Mon 30 Aug 2021 04:39:03 AM EDT restarting cinnamon.
Mon 30 Aug 2021 05:10:05 AM EDT restarting cinnamon.
Mon 30 Aug 2021 05:41:27 AM EDT restarting cinnamon.
Mon 30 Aug 2021 05:44:47 AM EDT restarting cinnamon.
Mon 30 Aug 2021 05:56:48 AM EDT restarting cinnamon.
Mon 30 Aug 2021 06:04:28 AM EDT restarting cinnamon.
Mon 30 Aug 2021 06:26:30 AM EDT restarting cinnamon.

Checking in ~/.xsession-errors I found:

Cjs-Message: 06:26:31.180: JS LOG: About to start Cinnamon

So while my alternative attempted to restart cinnamon while the screen-saver was active, it appears that it was only completely effective after the screeen-saver was deactivated.

I have added an additional diagnostic to my alternative script to include the return code from the call to xdotool that is simulating the restart cinnamon hot key signal to your cinnamon-killer process thinking it might shed some light on what is going on.  This might yield some further insight into the issue.

So my alternative script seems to be only partially success in restarting cinnamon while the screen-saver is active, but it appears to release some portion of the cinnamon memory or there would be many more entries in the restarts.log file.  It does avoid the error that is the subject of this issue, but that was not its complete purpose.

Your thoughts on this are welcome.

**Comment by mtwebster** [comment]
Your ctrl-alt-escape is being captured (deliberately) by the screensaver itself, which would be why restarts aren't happening until you unlock. I can't explain why you're not getting a restart attempt every SLEEP_TIME seconds. Perhaps some garbage collection is occurring in cinnamon, bringing you below the threshold? Maybe you could have your script print the memory used every timeout, regardless of whether it wants to restart or not.

There's a simple way to do a single test:

- Start your script
- In another terminal type `sleep 5 && cinnamon-dbus-command leakMemory 2048`, then immediately lock the screen.

You'll see that after those 5 seconds, the screensaver unlock dialog appears (this is the xdotool being captured).

Once you unlock the screen, sometime within your SLEEP_TIME you'll get the restart.

I added that `leakMemory` command specifically for testing the restart mechanism out.

I may have brought this up in the past, but I wrote an applet quite a while ago to help monitor memory use: https://github.com/mtwebster/Cinnamonitor - it's invaluable ( at least to me) when working on these types of issues.

Using the same leakMemory command, I can confirm that the restarts are successful using the patched cinnamon-screensaver.

**Comment by dhdurgee** [comment]
Aha! I had not thought about the screen-saver itself getting in the way of the hot key.  That certainly explains the delayed restart and might imply that I might see the same problem with the unpatched screen-saver that the memory-limit feature does.

Given this I will give your patched screen-saver a try with the memory-limit feature, although at present I can only get the memory-limit appropriate for my system via dconf-editor as the current GUI lower limit it to high for me.

If the cinnamon-dbus-command leakMemory 2048 is safe under any circumstances I would consider adding it to my script immediately before the xdotool call, but if it could be problematic unless the screen-saver is active I would need to condition it upon testing for screen-saver activity.  In either of these cases I suspect that without the patched screen-saver I would see the same problem as the memory-limit feature, which would render it less useful than the built-in feature.

Any schedule for when the fixed version will hit the repository?

I might check out your cinnamon memory tool, as perhaps if I knew which applets were leaking I could point it out to support in hopes of it being addressed in later versions.

**Comment by mtwebster** [comment]
Yes you can run cinnamon-dbus-command any time (it will run any dbus command that cinnamon supports, including restarting it (just run it without any arguments to see the commands)

**Comment by dhdurgee** [comment]
As expected with the patched stage.py installed the screen-saver did not crash when cinnamon restarted.  But this was unexpected:

2021.08.31 00:49:07 - Usage: 320 MB - Limit: 320 MB - Freq: 20 sec
2021.08.31 04:03:41 - Usage: 320 MB - Limit: 320 MB - Freq: 20 sec
2021.08.31 07:17:53 - Usage: 320 MB - Limit: 320 MB - Freq: 20 sec

Makes me wonder if the screen-saver is a source of the memory leakage I am seeing that causes me to need a restart.

**Comment by mtwebster** [comment]
Weird I'll test it out on my end later

**Comment by LinuxOnTheDesktop** [comment]
_A troubling datum_.I tried the patched stage.py file. After a while, I got one of the screensaver hangs; but this time running `cinnamon-unlock-desktop` from a tty no longer unlocked the screensaver. I had to do `sudo reboot`.

If the screensaver bug is not a high-priority one, might I suggest that it becomes so?

**Comment by Boored-Dude** [comment]
I had deactivated the screensaver, but had left the energy-saving-options alone.
After some time the screen was turned off by the energy-saving. And i come back to my notebook to see that cinnamon had crashed and was in fallback-mode.

See the atached file for the Coredump.
[CoreCump-Cinnamon-crash-afer-powersaving.txt](https://github.com/linuxmint/cinnamon-screensaver/files/7192015/CoreCump-Cinnamon-crash-afer-powersaving.txt)

The notebook where this happens was installed with Mint 20 Cinnamon an then updated to 20.1 and then 20.2.

On an other notebook has the "Somthing went wrong..."-Scenario only happend once. This was installed along time ago with Mint Cinnamon 19.1 and has been updated with every version up to 20.2.

On a PC this error has never happened. He was installed a long time ago with Mint Cinnamon 18.1 and has been updated with every version up to 20.2.

**Comment by LinuxOnTheDesktop** [comment]
This severe problem persists. Also, and as I noted above, the proposed workaround worsens things on my system. I know bugs can be hard to fix and resources are limited but it does feel as though we really should not have to put up with this.

**Comment by LinuxOnTheDesktop** [comment]
Having upgraded to a new kernel (a 5.11 one), this time I found that . . the screensaver would not even allow me to put the caret in the password box. The state of the screensaver is deplorable.

**Comment by LinuxOnTheDesktop** [comment]
One of my Mint PCs is affected by this bug. On that system, I encounter the bug (or rather that of #263) whether or not power management is set to turn off the monitor.

**Comment by 392781** [comment]
I am having this issue as well.  

Oddly enough, this only happens once I suspend the desktop and leave it like that for a few minutes.

If I suspend, and resume like a minute after, the screensaver works fine.

**Comment by mfreeman72** [comment]
Also happens to me on occasion since updating from 20.1 to 20.2. It's started happening more frequently in the last week or two. It went from happening about 2-3 times a week to happening almost daily. It's not happening every time my computer suspends or screensaver turns on, but it's frequent enough to be an annoyance.

**Comment by EvilSupahFly** [comment]
Since disabling power-saving while the screen saver is active, preventing
Cinnamon from turning off the monitors, I haven't had the issue.

------------------------------------------------------------
"Yo, yo. I'm real happy for ya, and I'm glad you can read, but first, I
wanna say that Seann has the best email signatures of all time. OF ALL
TIME!"
-- Kanye West, 2009 Email Signature Awards
------------------------------------------------------------

**Comment by mfreeman72** [comment]
I disabled "power saving" (I think - I can't find that exact wording in any of the settings), and the problem went away for a bit, but has come back. I'm getting this problem almost daily.

**Comment by alexveden** [comment]
The same problem, it happens after suspend boot or hibernation. I have a feeling this begun after upgrading kernel from 5.4 to 5.10

 Mint 20.2

Linux  5.10.0-1052-oem #54-Ubuntu SMP Tue Nov 23 09:06:13 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

System reports are empty, but here is what I'm getting in syslog.


```
ек 20 22:13:18 ubernote kernel: cinnamon-screen[202925]: segfault at 6174617a ip 00007f355ffed160 sp 00007ffd3644fb38 error 4 in libgobject-2.0.so.0.6400.6[7f355ffbf000+36000]
дек 20 22:13:18 ubernote kernel: Code: fc 03 00 00 77 3c 48 c1 e8 02 48 8d 15 b9 31 02 00 48 8b 04 c2 48 81 fe fc 03 00 00 76 31 48 83 e6 fc 45 31 c0 48 85 c0 74 0e <f6> 40 16 04 74 05 48 8>
дек 20 22:13:18 ubernote systemd[1]: Started Process Core Dump (PID 203241/UID 0).
дек 20 22:13:19 ubernote cinnamon-session[1236]: WARNING: t+144986,57250s: Detected that screensaver has left the bus
дек 20 22:13:19 ubernote systemd-coredump[203242]: Process 202925 (cinnamon-screen) of user 1000 dumped core.
                                                      
                                                      Stack trace of thread 202925:
                                                      #0  0x00007f355ffed160 g_type_check_instance_is_a (libgobject-2.0.so.0 + 0x3c160)
                                                      #1  0x00007f355ee2a2fa gdk_device_get_source (libgdk-3.so.0 + 0x342fa)
                                                      #2  0x00007f355ee48bec n/a (libgdk-3.so.0 + 0x52bec)
                                                      #3  0x00007f355ee4914e n/a (libgdk-3.so.0 + 0x5314e)
                                                      #4  0x00007f355ee2dd49 n/a (libgdk-3.so.0 + 0x37d49)
                                                      #5  0x00007f355ee2e983 n/a (libgdk-3.so.0 + 0x38983)
                                                      #6  0x00007f355ee2ee24 n/a (libgdk-3.so.0 + 0x38e24)
                                                      #7  0x00007f355ee53fd7 n/a (libgdk-3.so.0 + 0x5dfd7)
                                                      #8  0x00007f355ee26f4d n/a (libgdk-3.so.0 + 0x30f4d)
                                                      #9  0x00007f356006304e g_main_context_dispatch (libglib-2.0.so.0 + 0x5204e)
                                                      #10 0x00007f3560063400 n/a (libglib-2.0.so.0 + 0x52400)
                                                      #11 0x00007f35600634a3 g_main_context_iteration (libglib-2.0.so.0 + 0x524a3)
                                                      #12 0x00007f355fe07fe5 g_application_run (libgio-2.0.so.0 + 0xe2fe5)
                                                      #13 0x00007f3560d4eff5 n/a (libffi.so.7 + 0x6ff5)
                                                      #14 0x00007f3560d4e40a n/a (libffi.so.7 + 0x640a)
                                                      #15 0x00007f356017e0a5 n/a (_gi.cpython-38-x86_64-linux-gnu.so + 0x290a5)
                                                      #16 0x00007f356017525c n/a (_gi.cpython-38-x86_64-linux-gnu.so + 0x2025c)
                                                      #17 0x00000000005f57ee PyObject_Call (python3.8 + 0x1f57ee)
                                                      #18 0x000000000056cbfb _PyEval_EvalFrameDefault (python3.8 + 0x16cbfb)
                                                      #19 0x00000000005696da _PyEval_EvalCodeWithName (python3.8 + 0x1696da)
                                                      #20 0x00000000005f6403 _PyFunction_Vectorcall (python3.8 + 0x1f6403)
                                                      #21 0x000000000056b5e0 _PyEval_EvalFrameDefault (python3.8 + 0x16b5e0)
                                                      #22 0x00000000005696da _PyEval_EvalCodeWithName (python3.8 + 0x1696da)
                                                      #23 0x000000000068db17 PyEval_EvalCode (python3.8 + 0x28db17)
                                                      #24 0x000000000067eeb1 n/a (python3.8 + 0x27eeb1)
                                                      #25 0x000000000067ef2f n/a (python3.8 + 0x27ef2f)
                                                      #26 0x000000000067efd1 n/a (python3.8 + 0x27efd1)
                                                      #27 0x000000000067f377 PyRun_SimpleFileExFlags (python3.8 + 0x27f377)
                                                      #28 0x00000000006b7902 Py_RunMain (python3.8 + 0x2b7902)
                                                      #29 0x00000000006b7c8d Py_BytesMain (python3.8 + 0x2b7c8d)
                                                      #30 0x00007f3560b730b3 __libc_start_main (libc.so.6 + 0x270b3)
                                                      #31 0x00000000005fb12e _start (python3.8 + 0x1fb12e)
                                                      
                                                      Stack trace of thread 202928:
                                                      #0  0x00007f3560c61aff __GI___poll (libc.so.6 + 0x115aff)
                                                      #1  0x00007f356006336e n/a (libglib-2.0.so.0 + 0x5236e)
                                                      #2  0x00007f35600634a3 g_main_context_iteration (libglib-2.0.so.0 + 0x524a3)
                                                      #3  0x00007f355ad1a99d n/a (libdconfsettings.so + 0xa99d)
                                                      #4  0x00007f356008cad1 n/a (libglib-2.0.so.0 + 0x7bad1)
                                                      #5  0x00007f3560b32609 start_thread (libpthread.so.0 + 0x9609)
                                                      #6  0x00007f3560c6e293 __clone (libc.so.6 + 0x122293)
                                                      
                                                      Stack trace of thread 202927:
                                                      #0  0x00007f3560c61aff __GI___poll (libc.so.6 + 0x115aff)
                                                      #1  0x00007f356006336e n/a (libglib-2.0.so.0 + 0x5236e)
                                                      #2  0x00007f35600636f3 g_main_loop_run (libglib-2.0.so.0 + 0x526f3)
                                                      #3  0x00007f355fe43f8a n/a (libgio-2.0.so.0 + 0x11ef8a)
                                                      #4  0x00007f356008cad1 n/a (libglib-2.0.so.0 + 0x7bad1)
                                                      #5  0x00007f3560b32609 start_thread (libpthread.so.0 + 0x9609)
                                                      #6  0x00007f3560c6e293 __clone (libc.so.6 + 0x122293)
                                                      
                                                      Stack trace of thread 202926:
                                                      #0  0x00007f3560c61aff __GI___poll (libc.so.6 + 0x115aff)
                                                      #1  0x00007f356006336e n/a (libglib-2.0.so.0 + 0x5236e)
                                                      #2  0x00007f35600634a3 g_main_context_iteration (libglib-2.0.so.0 + 0x524a3)
                                                      #3  0x00007f35600634f1 n/a (libglib-2.0.so.0 + 0x524f1)
                                                      #4  0x00007f356008cad1 n/a (libglib-2.0.so.0 + 0x7bad1)
                                                      #5  0x00007f3560b32609 start_thread (libpthread.so.0 + 0x9609)
                                                      #6  0x00007f3560c6e293 __clone (libc.so.6 + 0x122293)
дек 20 22:13:19 ubernote systemd[1]: systemd-coredump@5-203241-0.service: Succeeded.

```

**Comment by okkindel** [comment]
Same problem, syslog:

```
Jan  3 09:10:05 k171195 cinnamon-screensaver-pam-helper: pam_ecryptfs: seteuid error
Jan  3 09:10:07 k171195 systemd[1]: NetworkManager-dispatcher.service: Succeeded.
Jan  3 09:10:15 k171195 org.cinnamon.ScreenSaver[15951]: Trying to connect to logind...
Jan  3 09:10:15 k171195 org.cinnamon.ScreenSaver[15951]: Starting screensaver...
Jan  3 09:10:15 k171195 org.cinnamon.ScreenSaver[15951]: Successfully using logind
Jan  3 09:10:15 k171195 cinnamon-session[6296]: WARNING: t+197,44883s: Detected that screensaver has left the bus
Jan  3 09:10:15 k171195 org.cinnamon.ScreenSaver[15951]: Error in sys.excepthook:
Jan  3 09:10:15 k171195 org.cinnamon.ScreenSaver[15951]: Original exception was:
Jan  3 09:10:15 k171195 org.cinnamon.ScreenSaver[15951]: Error in sys.excepthook:
Jan  3 09:10:15 k171195 org.cinnamon.ScreenSaver[15951]: Original exception was:
Jan  3 09:10:15 k171195 org.cinnamon.ScreenSaver[15951]: Error in sys.excepthook:
Jan  3 09:10:15 k171195 org.cinnamon.ScreenSaver[15951]: Original exception was:
```

i'm not able to suspend my computer at all

**Comment by kafido** [comment]
TL;DR:
"**couldn't grab keyboard**" might be the issue ...

Long story:
Same problem here, getting a screensaver segfault, pretty much the same as "alexveden" posted.
[screensaver-segfault.txt](https://github.com/linuxmint/cinnamon-screensaver/files/7810198/screensaver-segfault.txt)

I'm using Mint for years now and I'm experiencing this kind of problem since 20.2 at least,
if not 20.1 or even 20, on three completely different Laptops!
Now I'm running 20.3 - I was hoping this could have been solved in the meantime - but it still exists.

I also fully agree to "392781" - same here::
"Oddly enough, this only happens once I suspend the desktop and leave it like that for a few minutes.
If I suspend, and resume like a minute after, the screensaver works fine."

Sometimes, after wake up, the **keyboard is completely dead**, not even CTRL-ALT-F1 works,
however, mouse does work and ssh-ing in from another machine through network works also ...

So reading the comment from "Boored-Dude" got my attention:   
"Aug 22 11:18:56 Mintyi5 org.cinnamon.ScreenSaver[3235]: couldn't grab keyboard"
However, I haven't had this line in my syslog so far.

Is there anything I can do to help solving this?
Any additional debug log I can turn on?

**Comment by Boored-Dude** [comment]
this error had not occoured a long time for me. 
But today, after waking up my Notebook from Standby, it showed its ugly face...
These are the journal-entrys: 
`Jan 14 17:45:40 mint-a6 kernel: cinnamon-screen[63199]: segfault at 76677276 ip 00007fbf0b960160 sp 00007ffcd92e5818 error 4 in libgobject-2.0.so.0.6400.6[7fbf0b932000+36000]
Jan 14 17:45:40 mint-a6 kernel: Code: fc 03 00 00 77 3c 48 c1 e8 02 48 8d 15 b9 31 02 00 48 8b 04 c2 48 81 fe fc 03 00 00 76 31 48 83 e6 fc 45 31 c0 48 85 c0 74 0e <f6> 40 16 04 74 05 48 85 f6 75 35 45 31 c0 44 89 c0 c3 66 0f 1f 44
Jan 14 17:45:40 mint-a6 systemd[1]: Created slice system-systemd\x2dcoredump.slice.
Jan 14 17:45:40 mint-a6 systemd[1]: Started Process Core Dump (PID 64103/UID 0).
Jan 14 17:45:43 mint-a6 cinnamon-session[1423]: WARNING: t+19242,31368s: Detected that screensaver has left the bus
Jan 14 17:45:45 mint-a6 systemd-coredump[64105]: Process 63199 (cinnamon-screen) of user 1001 dumped core.
                                                 
                                                 Stack trace of thread 63199:
                                                 #0  0x00007fbf0b960160 g_type_check_instance_is_a (libgobject-2.0.so.0 + 0x3c160)
                                                 #1  0x00007fbf0a7b82fa gdk_device_get_source (libgdk-3.so.0 + 0x342fa)
                                                 #2  0x00007fbf0a7bcab8 n/a (libgdk-3.so.0 + 0x38ab8)
                                                 #3  0x00007fbf0a7bcd17 n/a (libgdk-3.so.0 + 0x38d17)
                                                 #4  0x00007fbf0a7e1fd7 n/a (libgdk-3.so.0 + 0x5dfd7)
                                                 #5  0x00007fbf0a7b4f4d n/a (libgdk-3.so.0 + 0x30f4d)
                                                 #6  0x00007fbf0b9d604e g_main_context_dispatch (libglib-2.0.so.0 + 0x5204e)
                                                 #7  0x00007fbf0b9d6400 n/a (libglib-2.0.so.0 + 0x52400)
                                                 #8  0x00007fbf0b9d64a3 g_main_context_iteration (libglib-2.0.so.0 + 0x524a3)
                                                 #9  0x00007fbf0b77afe5 g_application_run (libgio-2.0.so.0 + 0xe2fe5)
                                                 #10 0x00007fbf0c726ff5 n/a (libffi.so.7 + 0x6ff5)
                                                 #11 0x00007fbf0c72640a n/a (libffi.so.7 + 0x640a)
                                                 #12 0x00007fbf0baf70a5 n/a (_gi.cpython-38-x86_64-linux-gnu.so + 0x290a5)
                                                 #13 0x00007fbf0baee25c n/a (_gi.cpython-38-x86_64-linux-gnu.so + 0x2025c)
                                                 #14 0x00000000005f57ee PyObject_Call (python3.8 + 0x1f57ee)
                                                 #15 0x000000000056cbfb _PyEval_EvalFrameDefault (python3.8 + 0x16cbfb)
                                                 #16 0x00000000005696da _PyEval_EvalCodeWithName (python3.8 + 0x1696da)
                                                 #17 0x00000000005f6403 _PyFunction_Vectorcall (python3.8 + 0x1f6403)
                                                 #18 0x000000000056b5e0 _PyEval_EvalFrameDefault (python3.8 + 0x16b5e0)
                                                 #19 0x00000000005696da _PyEval_EvalCodeWithName (python3.8 + 0x1696da)
                                                 #20 0x000000000068db17 PyEval_EvalCode (python3.8 + 0x28db17)
                                                 #21 0x000000000067eeb1 n/a (python3.8 + 0x27eeb1)
                                                 #22 0x000000000067ef2f n/a (python3.8 + 0x27ef2f)
                                                 #23 0x000000000067efd1 n/a (python3.8 + 0x27efd1)
                                                 #24 0x000000000067f377 PyRun_SimpleFileExFlags (python3.8 + 0x27f377)
                                                 #25 0x00000000006b7902 Py_RunMain (python3.8 + 0x2b7902)
                                                 #26 0x00000000006b7c8d Py_BytesMain (python3.8 + 0x2b7c8d)
                                                 #27 0x00007fbf0c5450b3 __libc_start_main (libc.so.6 + 0x270b3)
                                                 #28 0x00000000005fb12e _start (python3.8 + 0x1fb12e)
                                                 
                                                 Stack trace of thread 63200:
                                                 #0  0x00007fbf0c633aff __GI___poll (libc.so.6 + 0x115aff)
                                                 #1  0x00007fbf0b9d636e n/a (libglib-2.0.so.0 + 0x5236e)
                                                 #2  0x00007fbf0b9d64a3 g_main_context_iteration (libglib-2.0.so.0 + 0x524a3)
                                                 #3  0x00007fbf0b9d64f1 n/a (libglib-2.0.so.0 + 0x524f1)
                                                 #4  0x00007fbf0b9ffad1 n/a (libglib-2.0.so.0 + 0x7bad1)
                                                 #5  0x00007fbf0c504609 start_thread (libpthread.so.0 + 0x9609)
                                                 #6  0x00007fbf0c640293 __clone (libc.so.6 + 0x122293)
                                                 
                                                 Stack trace of thread 63202:
                                                 #0  0x00007fbf0c633aff __GI___poll (libc.so.6 + 0x115aff)
                                                 #1  0x00007fbf0b9d636e n/a (libglib-2.0.so.0 + 0x5236e)
                                                 #2  0x00007fbf0b9d64a3 g_main_context_iteration (libglib-2.0.so.0 + 0x524a3)
                                                 #3  0x00007fbf0660199d n/a (libdconfsettings.so + 0xa99d)
                                                 #4  0x00007fbf0b9ffad1 n/a (libglib-2.0.so.0 + 0x7bad1)
                                                 #5  0x00007fbf0c504609 start_thread (libpthread.so.0 + 0x9609)
                                                 #6  0x00007fbf0c640293 __clone (libc.so.6 + 0x122293)
                                                 
                                                 Stack trace of thread 63201:
                                                 #0  0x00007fbf0c633aff __GI___poll (libc.so.6 + 0x115aff)
                                                 #1  0x00007fbf0b9d636e n/a (libglib-2.0.so.0 + 0x5236e)
                                                 #2  0x00007fbf0b9d66f3 g_main_loop_run (libglib-2.0.so.0 + 0x526f3)
                                                 #3  0x00007fbf0b7b6f8a n/a (libgio-2.0.so.0 + 0x11ef8a)
                                                 #4  0x00007fbf0b9ffad1 n/a (libglib-2.0.so.0 + 0x7bad1)
                                                 #5  0x00007fbf0c504609 start_thread (libpthread.so.0 + 0x9609)
                                                 #6  0x00007fbf0c640293 __clone (libc.so.6 + 0x122293)
                                                 
                                                 Stack trace of thread 64088:
                                                 #0  0x00007fbf0c63989d syscall (libc.so.6 + 0x11b89d)
                                                 #1  0x00007fbf0ba23746 g_cond_wait_until (libglib-2.0.so.0 + 0x9f746)
                                                 #2  0x00007fbf0b9a6581 n/a (libglib-2.0.so.0 + 0x22581)
                                                 #3  0x00007fbf0ba004ca n/a (libglib-2.0.so.0 + 0x7c4ca)
                                                 #4  0x00007fbf0b9ffad1 n/a (libglib-2.0.so.0 + 0x7bad1)
                                                 #5  0x00007fbf0c504609 start_thread (libpthread.so.0 + 0x9609)
                                                 #6  0x00007fbf0c640293 __clone (libc.so.6 + 0x122293)
`

**Comment by codexp** [comment]
This has happened to me too, with an fresh Linux Mint installation.

And monitor does not switch off, even though it is set up:
![image](https://user-images.githubusercontent.com/6502486/155505477-1fd9a8d8-b61d-4a7e-95b5-a5502ba77c5d.png)

Notebook's and external monitor stay all night on.

**Comment by Dude7017** [comment]
I have experienced the "Something Went Wong with the screensaver" issue twice now since a FRESH-from-the-ground-up install of Mint 20.2. BOTH times it has occurred AFTER an overnight SUSPEND of the system. (Never went into any "screen-saver" mode).... both times just was using the computer, got tired - bedtime - selected SUSPEND, and get the error upon waking the computer after overnight.

Checked the " crash report " as mentioned above, and find NO CRASH REPORT.

Very similar to what " 392781 " says above.

Here is my system info, in case of need:
[System Info Dump.zip](https://github.com/linuxmint/cinnamon-screensaver/files/8800195/System.Info.Dump.zip)

**Comment by mfreeman72** [comment]
There seems to have been an uptick in occurances. It's happening daily now, sometimes multiple times per day, on my laptop. I sure hope this gets fixed for Mint 21. It's an incredibly irritating and annoying problem.

**Comment by LinuxOnTheDesktop** [comment]
@mfreeman72 

It can seem that the best way to get the devs's attention is to post the problem on the beta bug tracker, when that tracker becomes available.

**Comment by KyleFromKitware** [comment]
I am currently dealing with #417 which seems similar to this.

**Comment by mfreeman72** [comment]
I haven't seen this problem since upgrading to Linux Mint 21! This makes me a happy boy! :)

**Comment by tsoud** [comment]
I had the same issue as the OP with the exact same error screen, right after upgrading to Mint 21 last night. I was using Mint 20.2 before, never had that issue. I will update if it happens again. Below is system info:
```
System:
  Kernel: 5.15.0-47-generic x86_64 bits: 64 compiler: gcc v: 11.2.0 Desktop: Cinnamon 5.4.12
    tk: GTK 3.24.33 wm: Mutter dm: LightDM Distro: Linux Mint 21 Vanessa base: Ubuntu 22.04 jammy
Machine:
  Type: Desktop Mobo: ASUSTeK model: PRIME X570-P v: Rev X.0x serial: <superuser required>
    UEFI: American Megatrends v: 2812 date: 11/04/2020
CPU:
  Info: 16-core model: AMD Ryzen 9 5950X bits: 64 type: MT MCP arch: Zen 3 rev: 0 cache:
    L1: 1024 KiB L2: 8 MiB L3: 64 MiB
  Speed (MHz): avg: 2335 high: 3593 min/max: 2200/5083 boost: enabled cores: 1: 2047 2: 3589
    3: 2053 4: 2192 5: 2195 6: 2194 7: 2193 8: 2195 9: 2049 10: 2054 11: 2050 12: 3593 13: 2078
    14: 2052 15: 3586 16: 2050 17: 2049 18: 3589 19: 2047 20: 2051 21: 2049 22: 2049 23: 2050
    24: 2123 25: 2676 26: 2048 27: 2051 28: 2047 29: 3593 30: 2053 31: 2049 32: 2050
    bogomips: 217203
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm
Graphics:
  Device-1: NVIDIA GA102 [GeForce RTX 3080] vendor: Gigabyte driver: nvidia v: 470.141.03 pcie:
    speed: 16 GT/s lanes: 16 bus-ID: 08:00.0 chip-ID: 10de:2206
  Device-2: Logitech Webcam C270 type: USB driver: snd-usb-audio,uvcvideo bus-ID: 3-3:2
    chip-ID: 046d:0825
  Display: x11 server: X.Org v: 1.21.1.3 driver: X: loaded: nvidia
    unloaded: fbdev,modesetting,nouveau,vesa gpu: nvidia display-ID: :0 screens: 1
  Screen-1: 0 s-res: 7936x2304 s-dpi: 156
  Monitor-1: DP-4 pos: right res: 4096x2304 dpi: 174 diag: 685mm (27")
  Monitor-2: HDMI-1 pos: primary,left res: 3840x2160 dpi: 163 diag: 685mm (27")
  OpenGL: renderer: NVIDIA GeForce RTX 3080/PCIe/SSE2 v: 4.6.0 NVIDIA 470.141.03
    direct render: Yes
```
Current screen saver and power settings are `turn off inactive screen: 2 hrs`, `suspend: never`, `delay before starting screensaver: 30 min`, `lock when put to sleep: Yes`, `lock after screensaver starts: Yes`.

**Comment by bolson22** [comment]
Es la primera vez que me sucede que aparece al volver a iniciar la PC luego de quedó en modo suspensión, aparece leyenda "Algo salio mal con el salvantallas". Me dice que para ir a la consola utilice "Control-Alt-F1" pero no pasa nada, probe con otras combinaicones de "Ctrl-Alt.." pero no logro salir de ahi ¿Alguna sugerencia? Mucha gracias

**Comment by tkrag** [comment]
I've had a similar issue where I can realiably make cinnamon freeze if the screen is locked and the monitors turn off. This happens bothj when powersaving turns the monitors off and when I just turn them off on their power buttons. If the screen is locked, the system freezes. Mouse cursor moves around but nothin else works. 


Swithing to tty and using the cinnamon-unlock-desktop command clears it up. 

This seems to only happen when one of my monitors is in portrait mode. If both are in landscape and therefore have identical resolution, the bug goes away. 

I'm running this on a Ryzen 7 with Radeon graphics.

**Comment by mtwebster** [comment]
Please have a look here: https://github.com/linuxmint/cinnamon-screensaver/issues/416#issuecomment-1256233029

Thanks

**Comment by ikunat33** [comment]
This is happening to me a lot.
I also have a Ryzen 7 with Radeon graphics. ROG Strix g513 (2022)

Every time cinnamon crashes, or brave or firefox dumps core, or my system just outright freezes up I can go into journalctl and scroll back to a point just moments before and there it is:

WARNING: t+46.38729s: Detected that screensaver has left the bus

followed by many things including:

Nov 01 19:33:44 user-xtc kernel: nvidia 0000:01:00.0: PCIe Bus Error: severity=Corrected, type=Data Link Layer, (Receiver ID)
Nov 01 19:33:44 user-xtc kernel: nvidia 0000:01:00.0:   device [10de:25a2] error status/mask=00000040/0000a000
Nov 01 19:33:44 user-xtc kernel: nvidia 0000:01:00.0:    [ 6] BadTLP                

I just killed a mint installation that was having tones of crashes after messages about the screensaver leaving the buse but I kept the logs.

I could not even log in for more that 2 min before the screensaver would leave the bus and it would crash.

Is there any headway with this?

**Comment by ikunat33** [comment]
I have fixed this issue and all my other problems with Linux crashing. It was directly related to the performance levels of my NVIDA GPU. Once I set PowerMizer to Prefer Maximum Performance everything stabilised and I have been running without issue.


See full explanation in page 3 of [MY POST ](https://forums.linuxmint.com/viewtopic.php?p=2392908#p2392908) on Linux Mint Forum.

**Comment by tuxut** [comment]
I have mint 21.2 installed on more than one machine with different hardware. On two of them I saw the cinnamon ScreenSaver crashing. I simply opend d-feed and then (Session Bus) clicking on org.cinnamon.ScreenSaver.
The text below the node says:
activatable: yes, pid: 13670, cmd: cinnamon-screensaver
Then I have only to simply wait for 30 seconds. The right field (Object path) closes and the the text "pid 13670, cmd: cinnamon-screensaver" disappears. When I click on or.cinnamon.ScreenSaver again, the pid has changed.
 
 syslog:
 Nov 14 22:01:11 dalex dbus-daemon[1372]: [session uid=1000 pid=1372] Successfully activated service 'org.cinnamon.ScreenSaver'
Nov 14 22:01:11 dalex org.cinnamon.ScreenSaver[136754]: Starting screensaver...
Nov 14 22:01:11 dalex org.cinnamon.ScreenSaver[136754]: Cinnamon Screensaver support not found in current theme - adding some...
Nov 14 22:01:11 dalex cinnamon-session[1373]: WARNING: t+55221,10493s: Detected that screensaver has appeared on the bus
Nov 14 22:01:11 dalex org.cinnamon.ScreenSaver[136754]: Successfully using logind
Nov 14 22:01:41 dalex cinnamon-session[1373]: WARNING: t+55251,10807s: Detected that screensaver has left the bus]

**Comment by SweeTeaRex** [comment]
I am experiencing this issue and cannot find a solution.

**Comment by squidder3** [comment]
> I have fixed this issue and all my other problems with Linux crashing. It was directly related to the performance levels of my NVIDA GPU. Once I set PowerMizer to Prefer Maximum Performance everything stabilised and I have been running without issue.
> 
> See full explanation in page 3 of [MY POST ](https://forums.linuxmint.com/viewtopic.php?p=2392908#p2392908) on Linux Mint Forum.

I personally wouldn't go about "fixing" the problem in this way. That definitely should not be needed, and I don't recommend anyone else doing that.

**Comment by EvilSupahFly** [comment]
> > I have fixed this issue and all my other problems with Linux crashing. It was directly related to the performance levels of my NVIDA GPU. Once I set PowerMizer to Prefer Maximum Performance everything stabilised and I have been running without issue.
> > See full explanation in page 3 of [MY POST ](https://forums.linuxmint.com/viewtopic.php?p=2392908#p2392908) on Linux Mint Forum.
> 
> I personally wouldn't go about "fixing" the problem in this way. That definitely should not be needed, and I don't recommend anyone else doing that.

Well, no, it **shouldn't** be needed. Normally. However, nVidia doesn't always play by the rules and users sometimes have to play fast and loose with the settings to achieve stability - something I've experienced far too many times. What was the alternative? Keep booting Windows to tweak the GPU then hope Linux boots? Use the open source driver and possibly not have access to some of your hardware's features or capabilities? If it works, it works. Presumably this issue was resolved at the source by nVidia with a driver update, of which there's been plenty in the last year.

**Comment by kafido** [comment]
From my experience this has NOTHING to do with NVIDIA. I had the same issue on multiple Devices (Notebooks) that don't even have a NVIDIA Card/Chip. Even AMD/Intel doesn't make a difference ...
See my comment above: https://github.com/linuxmint/cinnamon-screensaver/issues/392#issuecomment-1005139438
I've experienced this issue for some more years (!) with all subsequent versions of Linux Mint until I finally switched to Manjaro earlier this year. Did not have any similar issue since then on the exact same hardware!

**Comment by EvilSupahFly** [comment]
So your experience was different. Doesn't invalidate the "possible nVidia bug" claim since using an nVidia tool to change nVidia settings on nVidia hardware seems to have resolved @ikunat33's issue. In my case, the issue went away when the patch for the Cinnamon Screensaver was released. Aside from this one small thing, I've never had an issue running Mint on anything, outside of nVidia drivers not keeping pace with Kernel updates sometimes. My experience was different from yours. That's just how it is sometimes. That's the nature of Linux. 

```
¯\_(ツ)_/¯
```

**Comment by mtwebster** [comment]
> Then I have only to simply wait for 30 seconds. The right field (Object path) closes and the the text "pid 13670, cmd: cinnamon-screensaver" disappears. When I click on or.cinnamon.ScreenSaver again, the pid has changed.

This is normal behavior - the screensaver process exits after 30 seconds when not in use.

**Comment by squidder3** [comment]
> > > I have fixed this issue and all my other problems with Linux crashing. It was directly related to the performance levels of my NVIDA GPU. Once I set PowerMizer to Prefer Maximum Performance everything stabilised and I have been running without issue.
> > > See full explanation in page 3 of [MY POST ](https://forums.linuxmint.com/viewtopic.php?p=2392908#p2392908) on Linux Mint Forum.
> > 
> > 
> > I personally wouldn't go about "fixing" the problem in this way. That definitely should not be needed, and I don't recommend anyone else doing that.
> 
> Well, no, it **shouldn't** be needed. Normally. However, nVidia doesn't always play by the rules and users sometimes have to play fast and loose with the settings to achieve stability - something I've experienced far too many times. What was the alternative? Keep booting Windows to tweak the GPU then hope Linux boots? Use the open source driver and possibly not have access to some of your hardware's features or capabilities? If it works, it works. Presumably this issue was resolved at the source by nVidia with a driver update, of which there's been plenty in the last year.

I said that because it's obviously not a Nvidia specific issue seeing as people without Nvidia have the same issue. So ramping up your GPU 24/7 and lessening it's lifespan instead of finding the actual cause is a horrible idea. Period.

**Comment by sailnfool** [comment]
"Something went wrong" recovery instructions don't work.
after leaving the machine running for 3 days, I returned to this machine and found the recovery instructions.
CTRL-ALT-F1 does nothing.  I also tried CTRL-ALT-DEL and no response. I had to power cycle the machine to recover and lost 3 days of run-time.  Will attempt to reproduce.  System info for Linux Mint 22.1 attached

[renidea.info.txt](https://github.com/user-attachments/files/20217634/renidea.info.txt)

**Comment by bluezclouds** [comment]
Brand new, clean installation of Mint 22.1 on an AMD system and this is happening regularly. Since this problem hasn't been addressed in over 4 years I'm guessing posting system details is futile.

Has anyone found a solution or an alternate screensaver that works?

**Comment by Moderflator** [comment]
I have the same issue on a clean install of 22.1.  It happens quite often...once or more per day.  I access mint cinnamon 22.1 on a headless NUC via NoMachine.  As above, I cannot get to the shell to use desktop unlock.

The only thing that seems to work is a hard reboot.  Needless to say, this makes Mint's use as a headless anything impossible.

Please help.

**Comment by bluezclouds** [comment]
A work around for this bug is installing an HDMI dummy plug on the headless server. Not ideal, but better than a screensaver crash each time I connect.

**Comment by degreal** [comment]
Which settings to configure on fresh install so this error would never appear? Happaned on my all AMD system: AMD chipset, AMD 3400G CPU with AMD graphics. No NVidia in sight.

**Comment by ikunat33** [comment]
Hey degreal, I dont know if it was related to my nvidia card or my amd video. My system is a rog strix  G513RC and has both. I would mostly see nvidia in the crash report but amd showed up quite often as well. Until I selected "prefer max performance" in power management under system settings all sorts of things would dump core on me. I was just following the most likely scenario I could find based on the issues I was having and based on what I could find online until then.
The truth is that was the magic switch for me. My system still runs quiet with no fans unless I play a game or have a gazillion internet tabs open while doing other stuff as well. it rarely runs hot. right now my temps in Celsius are:
AMD Ryzen CPU: 47.1
AMD Graphics card: 46.0
NVME SSD: 26.9
CPU FAN: 2700 RPM
GPU FAN: 3100 RPM

which is pretty much in line with what I would get on my other power profiles. I have minecraft idling in the background right now while I browse and answer emails.

I have seen some posts here saying it is not a good idea to ramp up your graphics cards etc..  I am not seeing any increased temps or issues. I never turn my system off. I can not hear the fans at night when everything else is turned off. My system will even go into standby if I let it idol and resume without issue.
I just have a stable system since I chose prefer max performance in the linux power management. I am now running the latest linux mint and have not changed anything and all is good.
I dont touch the nvidia or amd settings at all.. just the linux power profile.
dont know if this will help but thought I would throw it out there just in case.

**Comment by adameus03** [comment]
Hi, I just  got the "Something went wrong with the screensaver" screen for the first time, FWIW this is a segfault that i found reported in dmesg logs:
```
[Thu Aug 27 16:17:04 2026] cinnamon-screen[2686721]: segfault at 7ffdaca38ff8 ip 00007f98ffb3836b sp 00007ffdaca39000 error 6 in libc.so.6[7f98ffac7000+156000] likely on CPU 1 (core 1, socket 0)
[Thu Aug 27 16:17:04 2026] Code: 0d fa 60 10 00 48 8d 15 53 68 10 00 bf 01 00 00 00 31 c0 48 8d 35 25 58 10 00 e8 70 7e fe ff 48 85 f6 0f 88 f7 05 00 00 41 57 <41> 56 41 55 41 54 49 89 f4 55 48 89 fd 53 48 8d 5e 17 48 83 ec 28
```
